debezium-embedded/src/main/java/io/debezium/embedded/async/ParallelSmtAndConvertAsyncConsumerProcessor.java

```diff
@@ -40,14 +39,20 @@ public class ParallelSmtAndConvertAsyncConsumerProcessor<R> extends AbstractReco
     @Override
     public void processRecords(final List<SourceRecord> records) throws Exception {
         LOGGER.debug("Thread {} is submitting {} records for processing.", Thread.currentThread().getName(), records.size());
-        final List<Future<Void>> recordFutures = new ArrayList<>(records.size());
-        records.stream().forEachOrdered(
-                r -> recordFutures.add(recordService.submit(new ProcessingCallables.TransformConvertConsumeRecord<>(r, transformations, convertor, consumer))));
+
+        final Future<Void>[] recordFutures = new Future[records.size()];
+        int arrayIndex = 0;
```

Comment:
I was thinking about this alternative too
```java
        Iterator<SourceRecord> recordsIterator = records.iterator();
        for (int i = 0; recordsIterator.hasNext(); i++) {
            recordFutures[i] = recordService.submit(new ProcessingCallables.TransformConvertConsumeRecord<>(r, transformations, convertor, consumer));
        }
```
choose the one you like more

Reply:
I think this one is actually better - it's 2 lines shorter (per processor) and I'd say also more idiomatic, so let's use this one. Thanks!